### PR TITLE
feat: handle logic to ofac restrict programs

### PR DESCRIPTION
**Pull request by AliAdnanSohail**
[Create a model that handles select program types based on select product Source to be automatically moved in Legal Review upon creation via CSV Ingestion or Django](https://2u-internal.atlassian.net/browse/PROD-3022)

## Description
We need to OFAC restrict specific program types of specific product source. So this PR will add logic to check if current program/degree object's program type exist in restricted program types, then it will be marked as OFAC restricted

### CODE REVIEW

File: course_discovery/apps/course_metadata/admin.py [hunk 1]
```diff
@@ -447,6 +447,9 @@ def response_change(self, request, obj):
 
     def save_model(self, request, obj, form, change):
         try:
+            if obj.type in obj.product_source.ofac_restricted_program_types.all():
+                obj.has_ofac_restrictions = True
+                obj.ofac_comment = f"Program type {obj.type.slug} is OFAC restricted for {obj.product_source.name}"
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/admin.py]: nit: I would suggest moving this message to some constant/settings.

File: course_discovery/apps/course_metadata/admin.py [hunk 2]
```diff
@@ -447,6 +447,9 @@ def response_change(self, request, obj):
 
     def save_model(self, request, obj, form, change):
         try:
+            if obj.type in obj.product_source.ofac_restricted_program_types.all():
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/admin.py]: perf/nit: We are fetching all types from DB. Instead, we can check the existence of obj.type in source's restricted program types.

File: course_discovery/apps/course_metadata/data_loaders/degrees_loader.py [hunk 3]
```diff
@@ -283,6 +283,10 @@ def _update_or_create_degree(
             defaults=data_dict
         )
 
+        if degree.product_source and program_type in degree.product_source.ofac_restricted_program_types.all():
+            degree.has_ofac_restrictions = True
+            degree.save()
+
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/data_loaders/degrees_loader.py]: Move ofac handling logic to models.py and use that method both here and in admin.

**Review by zawan-ila — APPROVED** [review]

File: course_discovery/apps/course_metadata/data_loaders/degrees_loader.py [hunk 4]
```diff
@@ -283,6 +283,10 @@ def _update_or_create_degree(
             defaults=data_dict
         )
 
+        if degree.product_source and program_type in degree.product_source.ofac_restricted_program_types.all():
+            degree.has_ofac_restrictions = True
+            degree.save()
```
Comment (Ali-D-Akbar) [inline course_discovery/apps/course_metadata/data_loaders/degrees_loader.py]: Are we not adding comments in case of degree?

**Review by Ali-D-Akbar — COMMENTED** [review]
A PR description would be nice to have which explains the approach and changes done in the PR. Approval once the comments are addressed.

File: course_discovery/apps/course_metadata/admin.py (hunk 1 above)
Comment (AliAdnanSohail) [reply course_discovery/apps/course_metadata/admin.py]: As we are using this text just here only, so I don't think we need to save it into constant

**Review by Ali-D-Akbar — APPROVED** [review]

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**